internal/utils/path_manager.go

```diff
@@ -43,6 +47,46 @@ func (p *PathManager) CniPath(flavour Flavour) (string, error) {
 	}
 }
 
+func (p *PathManager) VendorPluginSocket() string {
+	return p.wrap("/var/run/dpu-daemon/vendor-plugin/vendor-plugin.sock")
+}
+
 func (p *PathManager) wrap(path string) string {
 	return filepath.Join(p.rootDir, path)
 }
+
+// EnsureSocketDirExists makes sure that the socket being created is only accessible to root.
```

Comment:
I like how you moved this to `PathManager`

Reply:
Thanks!